dashboard/lib/sample_data.rb

```diff
@@ -0,0 +1,181 @@
+    unless user.nil?
+      user.sections.each do |section|
+        # Delete all students in each section
+        section.followers.each do |follower|
+          student_user = User.find_by_id(follower.student_user_id)
+          raise "Not a test user" unless student_user.name.include? "Codeberg"
+          UserGeo.where(user_id: student_user.id).destroy_all
+          student_user.really_destroy!
+        end
+        # Delete each section
+        section.really_destroy!
+      end
+      UserGeo.where(user_id: user.id).destroy_all
+      # Delete the existing test teacher
+      raise "Not a test user" unless user.name.include? "Codeberg"
+      user.really_destroy!
+    end
+    # Create the test teacher
+    create :teacher, {email: email, name: name,
+      password: TEST_ACCOUNT_PASSWORD, terms_of_service_version: 1}
+  end
+
+  # Generate a random gender choice with reasonable distributions
+  def self.random_gender
+    val = prng.rand(100)
+    gender = nil
+    case val
+    when 0..29
+      gender = "m"
+    when 30..59
+      gender = "f"
+    when 60..89
+      gender = nil
+    when 90..94
+      gender = "n"
+    when 95..99
+      gender = "o"
+    end
+    gender
+  end
+
+  # Create a test section
+  # Options is a hash with expected values:
+  #  :teacher - User object of teacher
+  #  :name - section name
+  #  :login_type - Section::LOGIN_TYPE_xxx value of login type for section
+  #  :grade - grade for section
+  #  :age_min - minimum age to generate for students
+  #  :age_max_inclusive - max age (inclusive) to generate for students
+  #  :script_name - name of script to assign for section
+  #  :num_students - number of students to generate for section
+  #  :use_imperfect_results - if true, generate some less than perfect
+  #    results. (CSF allows imperfect results, CSD and CSP do not.)
+  def self.create_section(options)
+    script = Script.get_from_cache(options[:script_name])
+    level_count = script.script_levels.count
+
+    # Create the section
+    section = create :section, {user: options[:teacher], name: options[:name],
+      login_type: options[:login_type], grade: options[:grade],
+      script: script}
+
+    # Create students in section
+    (0..options[:num_students] - 1).each do
+      # Choose random properties and create student
+      age_min = options[:age_min]
+      age_max_inclusive = options[:age_max_inclusive]
+
+      age = prng.rand(age_max_inclusive + 1 - age_min) + age_min
+      gender = random_gender
+      student_user = create :student, {age: age, gender: gender}
+
+      # Add student to section
+      create :follower, {section: section, student_user: student_user}
+
+      # Create random student progress.
+      pct_skipped = prng.rand(15)
+      pct_imperfect = options[:use_imperfect_results] ? prng.rand(40) + 25 +
+        pct_skipped : 0
+
+      max_level =
+        if prng.rand(100) < 90
+          (level_count.to_f * (prng.rand(0.6) + 0.2)).to_i
```

Comment:
Might be worth using a range for readability: `prng.rand(0.2..0.8)`

Reply:
Thx, done